### PR TITLE
Add es2015 modules support

### PR DESCRIPTION
**Pull request by a-tarasyuk**

### CODE REVIEW

**Comment by deregtd** [comment]
We'd like to test out the similar changes on subscribableevent before applying them in here, so this is likely to sit for a little bit with some people on vacation!

**Comment by a-tarasyuk** [comment]
@deregtd ok

**Comment by a-tarasyuk** [comment]
@deregtd 

In each repository(ReactXP, ReSub), I've created the branch `feature/test-es2015-modules` and replaced `ReSub@1.0.*` to `ReSub@feature/es2015-modules` in `package.json`. I think it a little bit simplify testing this PR. [ReactxXP/ImageList](https://github.com/a-tarasyuk/reactxp/tree/feature/test-resub-es2015-modules/samples/ImageList), [ReSub/async](https://github.com/a-tarasyuk/ReSub/tree/feature/test-es2015-modules/examples/async), [ReSub/counter](https://github.com/a-tarasyuk/ReSub/tree/feature/test-es2015-modules/examples/counter)

**ReSub/counter**(`ReSub#feature/es-2015`)
```shell
Hash: 3e77be439fa048292b88                                                              
Version: webpack 4.1.1
Time: 2332ms
     Asset       Size  Chunks             Chunk Names
 bundle.js    189 KiB       0  [emitted]  main
index.html  221 bytes          [emitted]  
```

**ReSub/counter**(`ReSub#1.0.8`)
```shell
Hash: b810ac8c71f68ec6bd07                                                              
Version: webpack 4.1.1
Time: 5680ms
     Asset       Size  Chunks             Chunk Names
 bundle.js    207 KiB       0  [emitted]  main
index.html  221 bytes          [emitted]  

```

**Comment by berickson1** [comment]
I think there's a couple merge conflicts from my other PR. Happy to merge this once we resolve those :)

**Comment by a-tarasyuk** [comment]
@berickson1 Ok, thanks, I'll resolve conflicts.

File: package.json [hunk 1]
```diff
@@ -5,65 +5,71 @@
...
+    "karma": "3.0.0",
     "karma-chrome-launcher": "2.2.0",
-    "karma-jasmine": "1.1.1",
-    "karma-jasmine-html-reporter": "1.0.0",
+    "karma-jasmine": "1.1.2",
+    "karma-jasmine-html-reporter": "1.3.1",
     "karma-sourcemap-loader": "0.3.7",
     "karma-spec-reporter": "0.0.32",
     "karma-webpack": "4.0.0-beta.0",
+    "npm-run-all": "4.1.3",
```
[the remark this answers is not in the document]
Comment (a-tarasyuk) [reply package.json]: this is helper to run npm commands, like so

```json
"build": "run-s clean:* tslint build:*",
```

**Pull request merged**